### PR TITLE
Add support for the CBOR data format

### PR DESCRIPTION
**Pull request by danielhenrymantilla**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -82,17 +83,30 @@
...
-//! You can safely process arbitrarily nested data without being exposed to
-//! stack overflows. Not even the Drop impl of our json `Value` type is
-//! recursive so you can safely nest them arbitrarily.
+//! In `miniserde::json` neither serialization nor deserialization involves
+//! recursion. You can safely process arbitrarily nested data without being
+//! exposed to stack overflows. Not even the Drop impl of our json `Value` type
+//! is recursive so you can safely nest them arbitrarily.
+//!
+//! On the other hand, `miniserde::cbor` deserialization **does use recursion**.
+//! It is capped, so that a deeply nested object (_e.g._, 256 layers) cause a
```
Comment (hamchapman) [inline src/lib.rs:99]:
```suggestion
//! It is capped, so that a deeply nested object (_e.g._, 256 layers) causes a
```

File: tests/cbor_std.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            let mut serialized = to_binary($s);
+            assert_eq!(to_vec(&expr).unwrap(), serialized);
+            let parsed: f64 = from_slice(&serialized[..]).unwrap();
+            if !expr.is_nan() {
+                assert_eq!(expr, parsed);
+            } else {
+                assert!(parsed.is_nan())
+            }
+
+            // let parsed: f64 = from_reader(&mut &serialized[..]).unwrap();
```
Comment (hamchapman) [inline tests/cbor_std.rs:33]: Are the commented out parts of this file intentionally left commented out for now?

**Review by hamchapman — APPROVED** [review]
❤️ 🚀  Awesome!

File: src/cbor/ser.rs [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+            assert_eq_hex!(&long_s[0..3], [0x59, 1, 0]);
+            assert_eq_hex!(&long_s[3..], &long_vec[..]);
+
+            // byte strings > 2^16 bytes have 5-byte headers
+            let very_long_vec = (0..65536).map(|i| (i & 0xFF) as u8).collect::<Vec<_>>();
+            let very_long_s = to_vec(&very_long_vec).unwrap();
+            assert_eq_hex!(&very_long_s[0..5], [0x5a, 0, 1, 0, 0]);
+            assert_eq_hex!(&very_long_s[5..], &very_long_vec[..]);
+
+            // byte strings > 2^32 bytes have 9-byte headers, but they take too much RAM
```
[the remark this answers is not in the document]
Comment (danielhenrymantilla) [reply src/cbor/ser.rs:537]: The tests were pulled from `serde_cbor` (and tweaked a bit with the `assert_eq_hex` for debugging, and with commented out code for parts that are not yet supported by `miniserde_ditto`). But thanks for pointing it out: I realize I forgot to add "credits" on that file

File: src/cbor/ser.rs [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+            assert_eq_hex!(vec, b"\x39\x01\x2b");
+            // i32
+            let vec = to_vec(&-23567997).unwrap();
+            assert_eq_hex!(vec, b"\x3a\x01\x67\x9e\x7c");
+            // u64
+            let vec = to_vec(&::std::u64::MAX).unwrap();
+            assert_eq_hex!(vec, b"\x1b\xff\xff\xff\xff\xff\xff\xff\xff");
+        }
+
+        // #[test]
```
[the remark this answers is not in the document]
Comment (danielhenrymantilla) [reply src/cbor/ser.rs:480]: Yeah, I'm not sure what we will be needing, but having these tests around can be helpful to spot the missing functionality 🙂

File: src/lib.rs (hunk 1 above)
Comment (danielhenrymantilla) [reply src/lib.rs:99]: Nothing gets by you🕵️, does it? 😄

**Pull request merged**